### ISSUE TITLE
npm run prod crashes app

### ISSUE THREAD

**Issue by cschra** [laravel-mix/laravel-mix#3302]
-   Laravel Mix Version: 6.0.49 
-   Node Version (`node -v`): 12.22.5
-   NPM Version (`npm -v`): 6.14.14
-   OS: CentOS 7

### Description:
I have an issue, that is related to Mix or Webpack: if I run `npm run dev` or `npm run watch` everything works, but as soon as I am trying to do `npm run prod`, the app is rendered and the page becomes blank after the js file is loaded.
The App is using Vue3 with composition API and script setup - As backend we use laravel with [nWidart/laravel-modules](https://github.com/nWidart/laravel-modules), hence the multiple js files.

My mix file looks as following:
```js
const path = require('path')
const mix = require('laravel-mix')

 require('laravel-mix-merge-manifest');
 mix.mergeManifest();

 mix.alias({
  '@': path.join(__dirname, 'resources/js')
 });

 mix.js('resources/js/app.js', 'public/js')
 .js('modules/module1/Resources/assets/js/module1.js', 'public/js')
 .js('modules/module2/Resources/assets/js/module2.js', 'public/js')
 .js('modules/module3/Resources/assets/js/module3.js', 'public/js')
 .js('modules/module4/Resources/assets/js/module4.js', 'public/js')
 .js('modules/module5/Resources/js/module5.js', 'public/js')
 .vue()
 .version()
 .postCss('resources/css/app.css', 'public/css', [
   require('tailwindcss'),
 ])

if (mix.inProduction()) {
  mix.sourceMaps()
}
```
### Steps To Reproduce:
run `npm run dev` or `npm run watch` (`mix`/`mix watch`) -> everything works
run `npm run prod` (`mix --production`) -> app crashes (log stays empty) and only a blank page is rendered

**Comment by Deivos** [comment]
We had something VERY similar in production. Our setup is with Vue 2 using:

- Laravel Mix Version: 6.0.19
- Node Version (node -v): 15.14.0
- NPM Version (npm -v): 7.7.6
- OS: Debian 9.13

The issue was present randomly sometimes our vue app would load we could not reliably reproduce the "blank" page and the only change in our code was some css and one string change from "white" to "black". Since reading about your issue I am wondering if this could be linked to a WebKit update. Since we had the issue near this date too.

**Comment by samgyamfi** [comment]
Make sure you do not have any <script></script> tags inside the #app block of your main layout blade file.
`<div id="app">Make sure there are not <script></script> in here. </div>`

**Comment by cschra** [comment]
I revisited this issue... we have no `<script>` tags inside our vue component/instances. However I found that our issue had to do with the `<script setup>` inside the vue files. Migrating to the "older" composition API with standard export solved the issue.

I believe this is an issue with the vue-loader/laravel mix linker/internal configuration. Interestingly I also setup my project with vite and ran into the same error. So this might be a general problem with vue (or vue loader idk)
I created a new issue with an in depth description of the problem and linked it to this one, so that interested people hopefully find this solution...
